fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -947,13 +949,14 @@ public static UnexpectedReadableException getUnexpectedReadableIfApplicable(@Nul
     }
 
     private static <T> Throwable findException(@Nullable Throwable ex, Class<T> classT) {
-        int allowedDepth = 20;
+        Set<Throwable> seenSet = new HashSet<>();
```

Comment:
You should probably use `IdentityHashMap` here, to avoid being affected by the potential `hashCode` implementations of the exceptions.

Reply:
✅